src/main/java/com/amihaiemil/docker/RtImages.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2018, Mihai Emil Andronache
+ * All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1)Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2)Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3)Neither the name of docker-java-api nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.amihaiemil.docker;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.stream.Collectors;
+import javax.json.Json;
+import javax.json.JsonObject;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+
+/**
+ * Runtime {@link Images}.
+ * @author George Aristy (george.aristy@gmail.com)
+ * @version $Id$
+ * @since 0.0.1
+ */
+final class RtImages implements Images {
+    /**
+     * Apache HttpClient which sends the requests.
+     */
+    private final HttpClient client;
+
+    /**
+     * Base URI.
+     */
+    private final URI baseUri;
+
+    /**
+     * Ctor.
+     * @param client The http client.
+     * @param uri The URI for this Images API.
+     */
+    RtImages(final HttpClient client, final URI uri) {
+        this.client = client;
+        this.baseUri = uri;
+    }
+
+    @Override
+    public Iterable<Image> iterate() throws IOException {
```

Comment:
@llorllale This is nice, for now. But in the future, I would like ``Images`` itself to be an ``Iterable<Image>``, so you can simply iterate over the Images like this:

```
for(Image img : docker.images()) {
 //...
}
```

So basically, this iterating logic will be inside a custom Iterator which would fetch them at before starting to iterate.

Reply:
@amihaiemil I like it. Will leave a puzzle.

Reply:
@amihaiemil btw, it will mean we will have to swallow an `IOException`. Are you ok with that?

Reply:
@llorllale Yes, I guess the tradeoff is worth it. And of course, we'll explain it in the javadoc.

Reply:
@amihaiemil right